pandas/core/indexing.py

```diff
@@ -2079,6 +2079,10 @@ def _validate_key(self, key, axis):
             arr = np.array(key)
             len_axis = len(self.obj._get_axis(axis))
 
+            if not np.issubdtype(arr.dtype, np.number):
+                raise IndexError(".iloc requires integer indexers, found "
+                                 "type {dtype}".format(dtype=arr.dtype))
```

Comment:
Would prefer if the passed indexer was shown in the error message as opposed to the dtype of the passed indexer.

i.e. `got 'foo'` instead of `found type <U3`